### PR TITLE
Added Azure.Automation.ManagedIdentity rule

### LINKED ISSUE #1074: Automation accounts should use managed identities for authentication
# Rule request

## Suggested rule change

As recently brought up in: https://azure.microsoft.com/en-us/updates/azure-automation-managed-identities-ga/, Managed identites for Automation accounts are GA and should be used. 

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.Automation/automationAccounts**

## Additional context

[Template reference](https://docs.microsoft.com/en-us/azure/templates/microsoft.automation/automationaccounts?tabs=bicep)

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fix #1074

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: docs/en/rules/Azure.Automation.ManagedIdentity.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+---
+severity: Important
+pillar: Security
+category: Identity and access management
+resource: Automation Service
```
Comment (ArmaanMcleod) [inline docs/en/rules/Azure.Automation.ManagedIdentity.md:5]: Wondering if we should use **Automation Account** for resource?

Comment (BernieWhite) [reply docs/en/rules/Azure.Automation.ManagedIdentity.md:5]: @ArmaanMcleod Yes, probably a good idea to change this.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Nice work! All good to merge. Simple cases might be good to do in YAML if they can be.

**Comment by ArmaanMcleod** [comment]
> @ArmaanMcleod Nice work! All good to merge. Simple cases might be good to do in YAML if they can be.

@BernieWhite Thanks 🙂. I renamed resource to Automation Account and converted rule to YAML. 

Looks like during rule generation the `Azure.Template.ExpressionLength` rule docs also got updated. Let me know if this looks ok to you.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Looks good to me.

**Pull request merged**